scipy/sparse/linalg/dsolve/linsolve.py

```diff
@@ -79,7 +79,16 @@ def _get_umf_family(A):
             % (f_type, i_type)
         raise ValueError(msg)
 
-    return family
+    if A.shape[0]*A.shape[1] > np.iinfo(np.int32).max:  # np.prod can overflow
```

Comment:
From the feedback in gh-8278: let's just make it switch to int64 unconditionally in all cases.
This is maybe about some internal umfpack data structure, and we should not really make assumptions about it here...